release_notes_presence_check/release_notes_presence_check_action.py

```diff
@@ -0,0 +1,168 @@
+    Class to handle the Release Notes Presence Check action.
+    """
+
+    def __init__(self) -> None:
+        """
+        Initialize the action with the required inputs.
+
+        @return: None
+        """
+        self.github_token: str = os.environ.get("INPUT_GITHUB_TOKEN", default="")
+        self.location: str = os.environ.get("INPUT_LOCATION", default="body")
+        self.title: str = os.environ.get("INPUT_TITLE", default="[Rr]elease [Nn]otes:")
+        self.skip_labels: list[str] = os.environ.get("INPUT_SKIP_LABELS", default="")
+        self.fails_on_error: bool = os.environ.get("INPUT_FAILS_ON_ERROR", "true").lower() == "true"
+
+        self.__validate_inputs()
+
+        self.pr_number: int = int(os.environ.get("INPUT_PR_NUMBER", default=""))
+        self.owner, self.repo_name = os.environ.get("INPUT_GITHUB_REPOSITORY", default="").split("/")
+
+    def run(self) -> None:
+        """
+        Run the action.
+
+        @return: None
+        """
+
+        # get PR information
+        repository = GitHubRepository(self.owner, self.repo_name, self.github_token)
+        pr_data = repository.get_pr_info(self.pr_number)
+
+        # check skip labels presence
+        labels = [label.get("name", "") for label in pr_data.get("labels", [])]
+        if self.skip_labels in labels:
+            logger.info("Skipping release notes check because '%s' label is present.", self.skip_labels)
+            self.write_output("true")
+            sys.exit(0)  # Exiting with code 0 indicates success but the action is skipped.
+
+        # check release notes presence in defined location
+        pr_body = pr_data.get("body", "")
+        if len(pr_body.strip()) == 0:
+            logger.error("Error: Pull request description is empty.")
+            self.handle_failure()
+
+        # Check if release notes tag is present
+        if not re.search(self.title, pr_body):
+            logger.error("Error: Release notes title '%s' not found in pull request body.", self.title)
+            self.handle_failure()
+
+        lines = pr_body.split("\n")
+        release_notes_start_index = None
+        for i, line in enumerate(lines):
+            if re.search(self.title, line):
+                release_notes_start_index = i + 1  # text after the tag line
+                break
+
+        if release_notes_start_index is None or release_notes_start_index >= len(lines):
+            logger.error("Error: No content found after the release notes tag.")
+            self.handle_failure()
+
+        text_below_release_notes = lines[release_notes_start_index:]
+        if not text_below_release_notes or not text_below_release_notes[0].strip().startswith(("-", "+", "*")):
+            logger.error("Error: No bullet list found directly under release notes tag.")
+            self.handle_failure()
+
+        self.write_output("true")
+        logger.info("Release Notes detected.")
+        sys.exit(0)
+
+    def write_output(self, valid_value) -> None:
```

Comment:
missing type hint.

Reply:
Added in commit 0dedf7f1b63e8e1e23f1f20e06f78cf22b698478.